Sources/StreamChatUI/Utils/KeyboardHandler/DefaultScrollViewKeyboardHandler.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+/// The component for handling keyboard events in a table view.
+open class DefaultScrollViewKeyboardHandler: KeyboardHandler {
+    public weak var scrollView: UIScrollView?
+
+    public init(scrollView: UIScrollView?) {
+        self.scrollView = scrollView
+    }
+
+    public func start() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillChangeFrame),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillChangeFrame),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+    }
+
+    // swiftlint:disable notification_center_detachment
+
+    public func stop() {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    // swiftlint:enable notification_center_detachment
```

Comment:
We should probably remove this rule if we prefer not to use it in some cases.

Reply:
Yeah we should eventually remove it

Reply:
Disabled it already 👍 I've also disabled `switch_case_alignment` which was very annoying and goes against Xcode's default formatting